src/transaction/log_replication.hpp

```diff
@@ -52,16 +59,20 @@ namespace cublog
       void redo_upto (cubthread::entry &thread_entry, const log_lsa &end_redo_lsa);
       template <typename T>
       void read_and_redo_record (cubthread::entry &thread_entry, LOG_RECTYPE rectype, const log_lsa &rec_lsa);
+      void wait_parallel_replication_idle ();
 
+      std::unique_ptr <cubthread::entry_task> m_daemon_task;
       cubthread::daemon *m_daemon = nullptr;
 
       log_lsa m_redo_lsa = NULL_LSA;
-      mutable std::mutex m_redo_mutex;
+      mutable std::mutex m_redo_lsa_mutex;
       mutable std::condition_variable m_redo_condvar;
```

Comment:
Then maybe this should be m_redo_lsa_condvar too.

Reply:
Done